File: spec/fizzbuzz_spec.rb
require './lib/fizzbuzz'

describe 'fizzbuzz' do 
  it 'returns "fizz" when passed 3' do
    expect(fizzbuzz(3)).to eq 'fizz'
    expect(fizzbuzz(9)).to eq 'fizz'
    end
  it 'returns "buzz" when passed 5' do
    expect(fizzbuzz(5)).to eq 'buzz'
    expect(fizzbuzz(20)).to eq 'buzz'
  end
  it 'gives an "Error" when passed a non integer or integer below 0' do
  expect(fizzbuzz(-2)).to eq "ERROR Give me an integer from zero to 100"
  expect(fizzbuzz("happy")).to eq "ERROR Give me an integer from zero to 100"
  expect(fizzbuzz('')).to eq "ERROR Give me an integer from zero to 100"
  expect(fizzbuzz(5.6)).to eq "ERROR Give me an integer from zero to 100"
  end
  it 'returns "fizzbuzz" when passed an integer divisible by 3 & 5' do
  expect(fizzbuzz(15)).to eq 'fizzbuzz'
  expect(fizzbuzz(30)).to eq 'fizzbuzz'
end
end

File: spec/fizzBuzzSpec.js
describe('FizzBuzz', function() {
  var fizzBuzz;

  beforeEach(function() {
    fizzBuzz = new FizzBuzz();
  });

  describe('when multiples 3 and 5', function() {
    it('displays FizBuzz', function() {
      expect(fizzBuzz.play(15)).toEqual('FizzBuzz');
    });

  it('displays FizBuzz', function() {
      expect(fizzBuzz.play(30)).toEqual('FizzBuzz');
    });
  });

  describe('when multiple of 5', function() {
    it('displays Buzz', function() {
      expect(fizzBuzz.play(5)).toEqual('Buzz');
    });

    it('displays Buzz', function() {
      expect(fizzBuzz.play(25)).toEqual('Buzz');
    });
  });

  describe('when multiple of 3', function() {
    it('displays Fizz', function() {
      expect(fizzBuzz.play(3)).toEqual('Fizz');
    });

    it('displays Fizz', function() {
      expect(fizzBuzz.play(9)).toEqual('Fizz');
    });
  });

  describe('when any other number', function() {
    it('displays the number itself', function() {
      expect(fizzBuzz.play(1)).toEqual(1);
    });
    it('displays the number itself', function() {
      expect(fizzBuzz.play(7)).toEqual(7);
    });
  });
});
